server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java

```diff
@@ -102,9 +103,10 @@ public void execute(SearchContext context) {
             SearchHit[] hits = new SearchHit[context.docIdsToLoadSize()];
             Map<String, Object> sharedCache = new HashMap<>();
 
+            SearchLookup lookup = context.getQueryShardContext().newSearchLookup();
```

Comment:
why can't we re-use the search lookup from the query shard context?

Reply:
@jtibshirani can talk more about this, but my understanding is that fetch wants a different context so things like explaining a script don't stomp what we're doing. Right now it uses a "clean" `SourceLookup` and I'm mostly just continuing that. `SourceLookup` is pretty much the only thing was state in the first place.

Reply:
ok it would be nice to chat about this, I think the changes you are making make sense after all, the only thing I don't love is having to expose this newSearchLookup public method which I foresee will be misused in the future, because it's there, handy, and people will wonder "which one do I need: newSearchLookup or lookup ?"